src/commands/data/create/file.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
+import { Messages, SfError } from '@salesforce/core';
+import { ContentVersion, file2CV } from '../../../api/file/fileToContentVersion.js';
+
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+const messages = Messages.loadMessages('@salesforce/plugin-data', 'data.create.file');
+
+type CDLCreateRequest = {
+  ContentDocumentId: string;
+  LinkedEntityId: string;
+  ShareType: string;
+};
+
+export default class DataCreateFile extends SfCommand<ContentVersion> {
+  public static readonly summary = messages.getMessage('summary');
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessages('examples');
+
+  public static readonly flags = {
+    'target-org': Flags.requiredOrg(),
+    'api-version': Flags.orgApiVersion(),
+    name: Flags.string({
```

Comment:
How about calling this flag "--new-name"?  Just to make it clear that, well, the name is new :).  Just a thought.

Reply:
what's the alternative interpretation that `-new-name` clears up by eliminating?

Reply:
I think when i wrote my comment I was thinking that --name on a command is typically used to identify something that already exists; but I just looked at all our commands that have --name, and it's to give something a name, like in this instance.  So ignore, --name is great!

Reply:
Thanks, @mshanemc for referring me back to the demo video regarding the `--name` flag.  That was very helpful.

I see where @jshackell-sfdc was coming from with the `--new-name` suggestion. I think she has a point. For me, it wasn't immediately apparent that setting `--name` was optional and would overwrite the local filename.

Am I correct in understanding that a successful file upload is the combination of the following?

1. Create a `ContentDocument` 
2. Create a `ContentDocumentLink`

If that's correct, is it possible that the `Title` field of the [ContentDocument](https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_contentdocument.htm) metadata type is the thing that is getting set when the file name is "overridden" by the `--name`flag?

If that's the case, maybe this flag should be `--title` instead of `--name`?

Reply:
Ha, funny -- I just realized that now, see my comment/screenshot below!  I like --title

Reply:
> Am I correct in understanding that a successful file upload is the combination of the following?
> 
> 1. Create a `ContentDocument`
> 2. Create a `ContentDocumentLink`

Not exactly.  
- If you're just uploading a file, that's a ContentVersion (which automatically creates its parent ContentDocument)
- the ContentDocumentLink is only used if you pass `--parent-id` that's what related the File to the Record
 
> If that's the case, maybe this flag should be `--title` instead of `--name`?
OK, title it is!

Reply:
fyi, the title also becomes the filename in the org (for example, if you open it or download it, etc).